src/bigint.rs

```diff
@@ -2188,7 +2188,11 @@ impl Integer for BigInt {
     /// Calculates the Lowest Common Multiple (LCM) of the number and `other`.
     #[inline]
     fn lcm(&self, other: &BigInt) -> BigInt {
-        BigInt::from_biguint(Plus, self.data.lcm(&other.data))
+        if self.is_zero() && other.is_zero() {
+            Self::zero()
+        } else {
+            BigInt::from_biguint(Plus, self.data.lcm(&other.data))
+        }
```

Comment:
Do we really need `BigInt` to check this condition? I would think that if `BigUint` does the right thing, `BigInt` should get it for free.

Reply:
You are correct. I thought the test had failed for `BigInt` when I added the test case.

Reply:
Looks like you added `BigUint` second, so I guess you did need this at first.